crates/ruff_linter/src/rules/flake8_pyi/snapshots/ruff_linter__rules__flake8_pyi__tests__PYI058_PYI058.pyi.snap

```diff
@@ -1,58 +1,215 @@
+3  4  | 
+--------------------------------------------------------------------------------
+14 15 |     import collections.abc
+15 16 | 
+16 17 |     class IteratorReturningSimpleGenerator3:
+17    |-        def __iter__(self) -> collections.abc.Generator: ...  # PYI058 (use `Iterator`)
+   18 |+        def __iter__(self) -> Iterator: ...  # PYI058 (use `Iterator`)
+18 19 | 
+19 20 | def scope():
+20 21 |     import collections.abc
+
+PYI058.pyi:24:13: PYI058 [*] Use `Iterator` as the return value for simple `__iter__` methods
    |
- 9 | class IteratorReturningSimpleGenerator2:
-10 |     def __iter__(self, /) -> collections.abc.Generator[str, Any, None]: ...  # PYI058 (use `Iterator`)
-   |         ^^^^^^^^ PYI058
-11 | 
-12 | class IteratorReturningSimpleGenerator3:
+23 |     class IteratorReturningSimpleGenerator4:
+24 |         def __iter__(self, /) -> collections.abc.Generator[str, Any, None]: ...  # PYI058 (use `Iterator`)
+   |             ^^^^^^^^ PYI058
+25 | 
+26 | def scope():
    |
+   = help: Convert the return annotation of your `__iter__` method to `Iterator`
 
-PYI058.pyi:13:9: PYI058 Use `Iterator` as the return value for simple `__iter__` methods
+ℹ Safe fix
+   1  |+from collections.abc import Iterator
+1  2  | def scope():
+2  3  |     from collections.abc import Generator
+3  4  | 
+--------------------------------------------------------------------------------
+21 22 |     from typing import Any
+22 23 | 
+23 24 |     class IteratorReturningSimpleGenerator4:
+24    |-        def __iter__(self, /) -> collections.abc.Generator[str, Any, None]: ...  # PYI058 (use `Iterator`)
+   25 |+        def __iter__(self, /) -> Iterator[str]: ...  # PYI058 (use `Iterator`)
+25 26 | 
+26 27 | def scope():
+27 28 |     import collections.abc
+
+PYI058.pyi:31:13: PYI058 [*] Use `Iterator` as the return value for simple `__iter__` methods
    |
-12 | class IteratorReturningSimpleGenerator3:
-13 |     def __iter__(self, /) -> collections.abc.Generator[str, None, typing.Any]: ...  # PYI058 (use `Iterator`)
-   |         ^^^^^^^^ PYI058
-14 | 
-15 | class AsyncIteratorReturningSimpleAsyncGenerator1:
+30 |     class IteratorReturningSimpleGenerator5:
+31 |         def __iter__(self, /) -> collections.abc.Generator[str, None, typing.Any]: ...  # PYI058 (use `Iterator`)
+   |             ^^^^^^^^ PYI058
+32 | 
+33 | def scope():
    |
+   = help: Convert the return annotation of your `__iter__` method to `Iterator`
+
+ℹ Safe fix
+   1  |+from collections.abc import Iterator
+1  2  | def scope():
+2  3  |     from collections.abc import Generator
+3  4  | 
+--------------------------------------------------------------------------------
+28 29 |     import typing
+29 30 | 
+30 31 |     class IteratorReturningSimpleGenerator5:
+31    |-        def __iter__(self, /) -> collections.abc.Generator[str, None, typing.Any]: ...  # PYI058 (use `Iterator`)
+   32 |+        def __iter__(self, /) -> Iterator[str]: ...  # PYI058 (use `Iterator`)
```

Comment:
This is a bug in `get_or_import_symbol` (it doesn't know that it can reuse `collections.abc` because it's a two-part module). Need to fix.

Reply:
I'll fix this separately, since the current fix isn't "wrong", it's just suboptimal.